### PR TITLE
Edit app metadata UX updates.

### PR DESCRIPTION
**Pull request by deanhannigan**
## Description
UX updates when altering the app name, url and icon in the builder.

- Updated the builder `Settings` screen by making the app metadata form immediately accessible instead of being in a modal.
- Clicking on the app title in the builder will now reveal the same form allowing you to edit the app anywhere in the builder.
- The edit app functionality has also been removed from the publishing dropdown.
- `Bug` - Fix for local app store getting squashed on app update

## Addresses
- https://linear.app/budibase/issue/BUDI-7933/allow-app-name-and-url-to-be-edited-inline

## Screenshots
Revamped metadata screen.
Screenshot 2024-05-14 at 12 24 39

Disabled when the app is published
Screenshot 2024-05-14 at 12 26 12

New edit UX on the app title.
Screenshot 2024-05-14 at 12 25 32

Same UX when the app is published
Screenshot 2024-05-14 at 12 26 00

## Launchcontrol
App metadata UX updates in the builder. Click on the title to alter your app details.

### CODE REVIEW

**Review by aptkingston — COMMENTED** [review]
Looks great Dean, nice one! Just a few suggestions around some small refactors and cleanup 👌

File: packages/bbui/src/Actions/position_dropdown.js [hunk 1]
```diff
@@ -155,6 +155,8 @@ export default function positionDropdown(element, opts) {
         applyXStrategy(Strategies.StartToEnd)
       } else if (align === "left-outside") {
         applyXStrategy(Strategies.EndToStart)
+      } else if (align === "center") {
+        applyXStrategy(Strategies.MidPoint)
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/bbui/src/Actions/position_dropdown.js:159]: Kudos to you, it was really intuitive!

File: packages/builder/src/components/common/UpdateAppForm.svelte [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      name,
+      url,
+      iconName,
+      iconColor,
+    })
+
+    setupValidation()
+  }
+
+  $: if (values && $values) {
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/builder/src/components/common/UpdateAppForm.svelte]: Yep agreed, I've updated the flow here.

File: packages/builder/src/components/common/UpdateAppForm.svelte [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    url: $appStore.url,
+    iconName: $appStore.icon?.name,
+    iconColor: $appStore.icon?.color,
+  }
+
+  // On app update, reset the state.
+  $: if (appMeta) {
+    edited = false
+    const { name, url, iconName, iconColor } = appMeta
+    values.set({
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/builder/src/components/common/UpdateAppForm.svelte:40]: Agreed. I've updated the initialisation flow here to make it a little less reactive to appStore changes and bring it in line with the validation changes you suggested

File: packages/builder/src/components/common/UpdateAppForm.svelte [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+      name,
+      url,
+      iconName,
+      iconColor,
+    })
+
+    setupValidation()
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/builder/src/components/common/UpdateAppForm.svelte]: Updated the flow around validation to remove unnecessary checks and limited the `setupValidation` to once upon initialisation.

File: packages/builder/src/components/common/UpdateAppForm.svelte [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+  let updating = false
+  let edited = false
+
+  $: filteredApps = $appsStore.apps.filter(app => app.devId == $appStore.appId)
+  $: app = filteredApps.length ? filteredApps[0] : {}
+  $: appDeployed = app?.status === AppStatus.DEPLOYED
+
+  $: appIdParts = app.appId ? app.appId?.split("_") : []
+  $: appId = appIdParts.slice(-1)[0]
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/builder/src/components/common/UpdateAppForm.svelte]: This was definitely a hangover from the old modal ux. It was used in both the portal and builder and could expect dev or prod Ids. Not necessary anymore and I've refactored it out.

File: packages/builder/src/components/common/UpdateAppTopNav.svelte [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      <Icon size="S" name="Edit" color={"var(--grey-7)"} />
+    </span>
+  </div>
+</div>
+
+<Popover
+  customZindex={998}
+  bind:this={formPopover}
+  align="center"
+  disabled={!isPublished}
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/builder/src/components/common/UpdateAppTopNav.svelte]: Nice catch! Also yep, I refactored the form elements out and forgot to clear out the old code.

**Review by aptkingston — APPROVED** [review]
LGTM! Really nice improvement. From testing it out, one tiny UX improvement (in my opinion) would just be closing the top popover after saving 👌

**Pull request merged**